packages/vats/tools/ibc-utils.js

```diff
@@ -1,11 +1,132 @@
-export const REMOTE_ADDR_RE =
-  /^(?<hops>\/ibc-hop\/[^/]+)*\/ibc-port\/(?<portID>[^/]+)\/(?<order>ordered|unordered)\/(?<version>[^/]+)$/s;
-harden(REMOTE_ADDR_RE);
 /** @typedef {`/${string}ibc-port/${string}/${'ordered' | 'unordered'}/${string}`} RemoteIbcAddress */
-
-export const LOCAL_ADDR_RE = /^\/ibc-port\/(?<portID>[-a-zA-Z0-9._+#[\]<>]+)$/;
 /** @typedef {`/ibc-port/${string}`} LocalIbcAddress */
 
+/**
+ * @import {IBCChannelID, IBCChannelOrdering,
+ *   IBCConnectionID, IBCPortID} from '../src/types.js';
+ */
+
+export const IBC_ADDR_RE =
+  /^(?<hops>\/ibc-hop\/[^/]+)*\/ibc-port\/(?<portID>[^/]+)(\/(?<order>ordered|unordered)\/(?<version>[^/]+)(\/ibc-channel\/(?<channelID>[^/]+))?)?$/s;
+harden(IBC_ADDR_RE);
+
+export const VALID_ORDERS = ['ordered', 'unordered'];
+harden(VALID_ORDERS);
+
+/** @param {string} raw */
+export const encodeSlashes = raw => raw.replaceAll('/', '\\x2f');
+harden(encodeSlashes);
+
+/** @param {string} encoded */
+export const decodeSlashes = encoded => encoded.replaceAll('\\x2f', '/');
+harden(decodeSlashes);
+
+export const validateIbcAddress = addr => {
+  const match = addr.match(IBC_ADDR_RE);
+  // .groups is to inform TS
+  if (!(match && match.groups)) {
+    throw TypeError(
+      `IBC address ${addr} must be '(/ibc-hop/CONNECTION)*/ibc-port/PORT/(ordered|unordered)/VERSION(/ibc-channel/CHANNEL_ID)?'`,
+    );
+  }
+  return match;
+};
+
+/**
+ * @param {string} addr
+ * @returns {{
+ *   hops: IBCConnectionID[];
+ *   portID: IBCPortID;
+ *   order?: IBCChannelOrdering;
+ *   version?: string;
+ *   channelID?: IBCChannelID;
+ * }}
+ */
+export const decodeIbcEndpoint = addr => {
```

Comment:
I suppose this function is constrained by some interchain spec. I suggest we cite it in the docstring.

Reply:
Done.